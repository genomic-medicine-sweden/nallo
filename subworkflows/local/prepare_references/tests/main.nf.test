nextflow_workflow {

    name "Test Workflow PREPARE_REFERENCES"
    script "../main.nf"
    config "./nextflow.config"
    workflow "PREPARE_REFERENCES"

    test("unzip fasta, untar vep cache") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'hg38' ],
                    file(params.pipelines_testdata_base_path + 'reference/hg38.test.fa.gz', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'vep_cache' ],
                    file(params.pipelines_testdata_base_path + 'reference/vep_cache_test_data.tar.gz', checkIfExists:true)
                ])
                input[2] = true
                input[3] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

    }

    test("don't unzip fasta, untar vep cache") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'hg38' ],
                    file(params.pipelines_testdata_base_path + 'reference/hg38.test.fa', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'vep_cache' ],
                    file(params.pipelines_testdata_base_path + 'reference/vep_cache_test_data.tar.gz', checkIfExists:true)
                ])
                input[2] = false
                input[3] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

    }

    test("unzip fasta, untar vep cache -stub") {

        options "-stub"

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'hg38' ],
                    file(params.pipelines_testdata_base_path + 'reference/hg38.test.fa.gz', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'vep_cache' ],
                    file(params.pipelines_testdata_base_path + 'reference/vep_cache_test_data.tar.gz', checkIfExists:true)
                ])
                input[2] = true
                input[3] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

    }

    test("don't unzip fasta, don't untar vep cache -stub") {

        options "-stub"
        tag "xy"
        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'hg38' ],
                    file(params.pipelines_testdata_base_path + 'reference/hg38.test.fa', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id: 'vep_cache' ],
                    file("$projectDir/.devcontainer", checkIfExists:true)
                ])
                input[2] = false
                input[3] = false
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }

    }

}
