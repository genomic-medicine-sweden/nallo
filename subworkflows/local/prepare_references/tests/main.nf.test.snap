{
    "unzip fasta, untar vep cache": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,066634e80999105a03a3739b9b9133ca"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,fcb4c442b23f0477b270a8f2c7c2a353"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa:md5,a90ab1e088326745c2ae622afc6f197c"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            [
                                [
                                    "._info.txt:md5,3443c3a4577896a52f0105597567d263",
                                    [
                                        "1-1000000.gz:md5,f87d70049f19c7b7c19dee85499287f4",
                                        "1-1000000_reg.gz:md5,761c580dbb84ba81535ebc996af9c47f",
                                        "1-1000000_var.gz:md5,9c7c5960f7f87846083b39727a30b548"
                                    ],
                                    [
                                        "3000001-4000000.gz:md5,35abc3e1f0b356e88e4b674f54deddf3",
                                        "3000001-4000000_reg.gz:md5,2e75f25e392622b9be69bc24465a651c"
                                    ],
                                    [
                                        "140000001-141000000.gz:md5,d0bbdc9fd851d90cde9de7f4ffa7a38b",
                                        "140000001-141000000_reg.gz:md5,2fdef236fbb852e8ea155836e20ce994",
                                        "140000001-141000000_var.gz:md5,7d8ea7ee33e425cd4c1515a850e2b663"
                                    ],
                                    "info.txt:md5,0602ec9c683f5de9b058d9959ae31d18"
                                ]
                            ]
                        ]
                    ]
                ],
                "4": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,53ba80d1edc3452084e1ff487d1232ee",
                    "versions.yml:md5,7479477c0c4b6eb043c58bc40ee41198",
                    "versions.yml:md5,84898ecb507328de243b556f1327a78c"
                ],
                "fai": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,fcb4c442b23f0477b270a8f2c7c2a353"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa:md5,a90ab1e088326745c2ae622afc6f197c"
                    ]
                ],
                "mmi": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,066634e80999105a03a3739b9b9133ca"
                    ]
                ],
                "vep_resources": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            [
                                [
                                    "._info.txt:md5,3443c3a4577896a52f0105597567d263",
                                    [
                                        "1-1000000.gz:md5,f87d70049f19c7b7c19dee85499287f4",
                                        "1-1000000_reg.gz:md5,761c580dbb84ba81535ebc996af9c47f",
                                        "1-1000000_var.gz:md5,9c7c5960f7f87846083b39727a30b548"
                                    ],
                                    [
                                        "3000001-4000000.gz:md5,35abc3e1f0b356e88e4b674f54deddf3",
                                        "3000001-4000000_reg.gz:md5,2e75f25e392622b9be69bc24465a651c"
                                    ],
                                    [
                                        "140000001-141000000.gz:md5,d0bbdc9fd851d90cde9de7f4ffa7a38b",
                                        "140000001-141000000_reg.gz:md5,2fdef236fbb852e8ea155836e20ce994",
                                        "140000001-141000000_var.gz:md5,7d8ea7ee33e425cd4c1515a850e2b663"
                                    ],
                                    "info.txt:md5,0602ec9c683f5de9b058d9959ae31d18"
                                ]
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,53ba80d1edc3452084e1ff487d1232ee",
                    "versions.yml:md5,7479477c0c4b6eb043c58bc40ee41198",
                    "versions.yml:md5,84898ecb507328de243b556f1327a78c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-07-29T13:59:21.486640187"
    },
    "unzip fasta, untar vep cache -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            [
                                
                            ]
                        ]
                    ]
                ],
                "4": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,53ba80d1edc3452084e1ff487d1232ee",
                    "versions.yml:md5,586c5c7ffbdb5170c2a4e0af8f7a14aa",
                    "versions.yml:md5,7479477c0c4b6eb043c58bc40ee41198"
                ],
                "fai": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "mmi": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vep_resources": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            [
                                
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,53ba80d1edc3452084e1ff487d1232ee",
                    "versions.yml:md5,586c5c7ffbdb5170c2a4e0af8f7a14aa",
                    "versions.yml:md5,7479477c0c4b6eb043c58bc40ee41198"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-07-29T14:43:36.069166765"
    },
    "don't unzip fasta, don't untar vep cache -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "hg38"
                        },
                        "/genomic-medicine-sweden/test-datasets/9e0f40af8fb3ab7f8a3daecaeb03a0acda1e464e/reference/hg38.test.fa"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            "devcontainer.json:md5,5d0fb65a90fae7521d3d2cd952528f71"
                        ]
                    ]
                ],
                "4": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,586c5c7ffbdb5170c2a4e0af8f7a14aa"
                ],
                "fai": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "hg38"
                        },
                        "/genomic-medicine-sweden/test-datasets/9e0f40af8fb3ab7f8a3daecaeb03a0acda1e464e/reference/hg38.test.fa"
                    ]
                ],
                "mmi": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vep_resources": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            "devcontainer.json:md5,5d0fb65a90fae7521d3d2cd952528f71"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,586c5c7ffbdb5170c2a4e0af8f7a14aa"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-08T14:02:28.319340933"
    },
    "don't unzip fasta, untar vep cache": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,066634e80999105a03a3739b9b9133ca"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,fcb4c442b23f0477b270a8f2c7c2a353"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "hg38"
                        },
                        "/genomic-medicine-sweden/test-datasets/9e0f40af8fb3ab7f8a3daecaeb03a0acda1e464e/reference/hg38.test.fa"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            [
                                [
                                    "._info.txt:md5,3443c3a4577896a52f0105597567d263",
                                    [
                                        "1-1000000.gz:md5,f87d70049f19c7b7c19dee85499287f4",
                                        "1-1000000_reg.gz:md5,761c580dbb84ba81535ebc996af9c47f",
                                        "1-1000000_var.gz:md5,9c7c5960f7f87846083b39727a30b548"
                                    ],
                                    [
                                        "3000001-4000000.gz:md5,35abc3e1f0b356e88e4b674f54deddf3",
                                        "3000001-4000000_reg.gz:md5,2e75f25e392622b9be69bc24465a651c"
                                    ],
                                    [
                                        "140000001-141000000.gz:md5,d0bbdc9fd851d90cde9de7f4ffa7a38b",
                                        "140000001-141000000_reg.gz:md5,2fdef236fbb852e8ea155836e20ce994",
                                        "140000001-141000000_var.gz:md5,7d8ea7ee33e425cd4c1515a850e2b663"
                                    ],
                                    "info.txt:md5,0602ec9c683f5de9b058d9959ae31d18"
                                ]
                            ]
                        ]
                    ]
                ],
                "4": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,7479477c0c4b6eb043c58bc40ee41198",
                    "versions.yml:md5,84898ecb507328de243b556f1327a78c"
                ],
                "fai": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.fa.fai:md5,fcb4c442b23f0477b270a8f2c7c2a353"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "hg38"
                        },
                        "/genomic-medicine-sweden/test-datasets/9e0f40af8fb3ab7f8a3daecaeb03a0acda1e464e/reference/hg38.test.fa"
                    ]
                ],
                "mmi": [
                    [
                        {
                            "id": "hg38"
                        },
                        "hg38.test.mmi:md5,066634e80999105a03a3739b9b9133ca"
                    ]
                ],
                "vep_resources": [
                    [
                        {
                            "id": "vep_cache"
                        },
                        [
                            [
                                [
                                    "._info.txt:md5,3443c3a4577896a52f0105597567d263",
                                    [
                                        "1-1000000.gz:md5,f87d70049f19c7b7c19dee85499287f4",
                                        "1-1000000_reg.gz:md5,761c580dbb84ba81535ebc996af9c47f",
                                        "1-1000000_var.gz:md5,9c7c5960f7f87846083b39727a30b548"
                                    ],
                                    [
                                        "3000001-4000000.gz:md5,35abc3e1f0b356e88e4b674f54deddf3",
                                        "3000001-4000000_reg.gz:md5,2e75f25e392622b9be69bc24465a651c"
                                    ],
                                    [
                                        "140000001-141000000.gz:md5,d0bbdc9fd851d90cde9de7f4ffa7a38b",
                                        "140000001-141000000_reg.gz:md5,2fdef236fbb852e8ea155836e20ce994",
                                        "140000001-141000000_var.gz:md5,7d8ea7ee33e425cd4c1515a850e2b663"
                                    ],
                                    "info.txt:md5,0602ec9c683f5de9b058d9959ae31d18"
                                ]
                            ]
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,154e23eb131667048a63be0f683d6a92",
                    "versions.yml:md5,7479477c0c4b6eb043c58bc40ee41198",
                    "versions.yml:md5,84898ecb507328de243b556f1327a78c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-07-29T14:06:23.16468683"
    }
}